uint32 distance1
uint32 distance2
uint32 distance3
